pubsub/mqtt/mqtt.go

```diff
@@ -213,34 +213,23 @@ func (m *mqttPubSub) startSubscription(ctx context.Context) error {
 		// for backwards-compatibility; see: https://github.com/dapr/components-contrib/pull/2104
 		consumerClientID += "-consumer"
 	}
-	connCtx, connCancel := context.WithTimeout(ctx, defaultWait)
-	c, err := m.connect(connCtx, consumerClientID)
-	connCancel()
+
+	readyCh := make(chan struct{})
+	c, err := m.connect(ctx, consumerClientID, true, readyCh)
 	if err != nil {
 		return err
 	}
-	m.consumer = c
-
-	subscribeTopics := make(map[string]byte, len(m.topics))
-	for k := range m.topics {
-		subscribeTopics[k] = m.metadata.qos
-	}
-
-	token := m.consumer.SubscribeMultiple(
-		subscribeTopics,
-		m.onMessage(ctx),
-	)
-	subscribeCtx, subscribeCancel := context.WithTimeout(m.ctx, defaultWait)
-	defer subscribeCancel()
+	// This is to wait for the subscriptions to be enabled, which happen in an async callback
+	// We do this after checking for errors returned by m.connect
 	select {
-	case <-token.Done():
-		// Subscription went through
-	case <-subscribeCtx.Done():
-		return subscribeCtx.Err()
-	}
-	if err := token.Error(); err != nil {
-		return fmt.Errorf("mqtt error from subscribe: %v", err)
+	// Double the wait time here as we've seen that some brokers just need a bit more time
+	case <-time.After(defaultWait * 2):
```

Comment:
Is it correct to double the defaultTime here OR increase/double the default wait time in metadata.go itself?